features/google/src/model/google-account.js

```diff
@@ -0,0 +1,99 @@
+//TODO: Use DataAccount super class
+export default class GoogleAccount {
+    constructor() {
+        this._importingResults = [];
+        this._importedFileNames = [];
+
+        this._name = "";
+        this._preferredLanguage = [];
+        this._analyses = {};
+        this._reports = {};
+        this._analysesExecutionResults = [];
+        this._placeVisits = [];
+        this._activitySegments = [];
+        this._activities = [];
+        this._pathNames = [];
+    }
+
+    get analyses() {
+        return this._analyses;
+    }
+
+    get reports() {
+        return this._reports;
+    }
+
+    get importedFileNames() {
+        return this._importedFileNames;
+    }
+
+    addImportedFileName(fileName) {
+        this._importedFileNames.push(fileName);
+    }
+
+    get importingResults() {
+        return this._importingResults;
+    }
+
+    set importingResults(importingResults) {
+        this._importingResults = importingResults;
```

Comment:
Are we going to allow to set this outside the constructor? How will we maintain consistency with the execution results?

Reply:
Importers set these attributes, that's their whole point.

Reply:
I guessed as much, so bear with me here. We're apparently building an empty object initially. This object is handled around and "filled" with different attributes, but we can't be sure those attributes exist or not a a certain point in time. There's no internal logic that tells us when we have a valid object, up to an including initial or raw import. In general, objects should be valid from the moment they're created and onwards; instead, we're kinda creating an "incomplete" object that's going to be "filled" with different things. At the end of the day, we're simply using this object as a shelf for different, and seemingly unrelated, data analysis.

Reply:
I get what you are saying, but it's a feature of this isn't it? Attributes don't need to be filled eg. when the data is not there for this. So most of the time it is going to be incomplete